packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -53,6 +53,14 @@ export class CoinbaseWalletProvider extends ProviderEventEmitter implements Prov
             storeSignerType(signerType);
             break;
           }
+          case 'wallet_sendCalls':
+          case 'wallet_sign': {
+            const ephemeralSigner = this.initSigner('scw');
+            await ephemeralSigner.handshake({ method: 'coinbase_handshake' }); // exchange session keys
+            const result = await ephemeralSigner.request(args); // send diffie-hellman encrypted request
+            await ephemeralSigner.cleanup(); // clean up (rotate) the ephemeral session keys
+            return result as T;
+          }
```

Comment:
we are going to do similar thing for `wallet_connect`, just that the signer won't be ephemeral but will have some effect in SCWSigner.handshake